Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/cmd/get.go

```diff
@@ -289,9 +291,29 @@ func RunGet(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string
 	// use the default printer for each object
 	printer = nil
 	var lastMapping *meta.RESTMapping
+	var withKind bool = allKinds
 	w := kubectl.GetNewTabWriter(out)
 	defer w.Flush()
 
+	// determine if printing multiple kinds of
+	// objects and enforce "show-kinds" flag if so
+	for ix := range objs {
+		var mapping *meta.RESTMapping
+		if sorter != nil {
+			mapping = infos[sorter.OriginalPosition(ix)].Mapping
+		} else {
+			mapping = infos[ix].Mapping
+		}
+
+		// display "kind" column only if we have mixed resources
+		if lastMapping != nil && mapping.Resource != lastMapping.Resource {
+			withKind = true
+		}
+		lastMapping = mapping
+	}
+
+	lastMapping = nil
```

Comment:
This is calling out for a refactor / move it to another function instead of recycling variables.

The following `for` loop below could easily become a function, with a name explaining what it does, and it would return a `[]error`, such that the last lines would read something like:

```go
	// ...
	if mustPrintWithKinds(...) { // mustPrintWithKinds replaces yet another for-loop bloating the over-bloated RunGet.
		printer.EnsurePrintWithKind()  // this method doesn't exist yet, but would set PrintOptions.WithKind to true.
	}
	return utilerrors.NewAggregate(printObjects(printer, ...)) // printObjects replaces the huge final for-loop
}
```